### PR TITLE
fix: refactor ms-authentication to use logging client to log errors/info

### PR DESCRIPTION
**Pull request by joshua-silverio**
Partial Closes:#10
Closes:#15
Signed-off-by: intel <joshua.silverio@intel.com>

### CODE REVIEW

File: ms-authentication/routes/controller.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"fmt"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+)
+
+type Controller struct {
+	lc      logger.LoggingClient
```
Comment (jim-wang-intel) [inline ms-authentication/routes/controller.go:14]: this seems redundant. you can get loggingClient from interfaces.ApplicationService.LoggingClient() call

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
remove lc inside the Controller's struct as it can be obtained from `interfaces.ApplicationService`

File: ms-authentication/routes/controller.go (hunk 1 above)
Comment (joshua-silverio) [reply ms-authentication/routes/controller.go]: removed

**Review by jim-wang-intel — APPROVED** [review]
LGTM

File: ms-authentication/routes/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			mockAppService := &mocks.ApplicationService{}
+			if !tt.failAddRoute {
+				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+					Return(nil)
+			} else {
+				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+					Return(fmt.Errorf("fail"))
+			}
+
+			c := &Controller{
```
Comment (lenny-intel) [inline ms-authentication/routes/controller_test.go:43]: Use NewController()

File: ms-authentication/routes/controller.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	"fmt"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+)
+
+type Controller struct {
+	service interfaces.ApplicationService
+}
+
+func NewController(service interfaces.ApplicationService) Controller {
```
Comment (lenny-intel) [inline ms-authentication/routes/controller.go:16]: Just pass the logging client, since that is the only dependency that needs to be injected.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ms-authentication/routes/controller.go (hunk 3 above)
Comment (joshua-silverio) [reply ms-authentication/routes/controller.go:16]: I think the service client is needed for the AddAllRoutes function, so b/c service has lc Jim suggested just using service

Comment (lenny-intel) [reply ms-authentication/routes/controller.go:16]: Yes! I missed that usage.

File: ms-authentication/routes/controller.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	return Controller{
+		service: service,
```
Comment (lenny-intel) [inline ms-authentication/routes/controller.go:18]: Common practice is to also set the logger so each logging call doesn't have to go thru the service API
```suggestion
		service: service,
		lc: service.LoggerClient(),
```

File: ms-authentication/routes/controller.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+package routes
+
+import (
+	"fmt"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+)
+
+type Controller struct {
+	service interfaces.ApplicationService
```
Comment (lenny-intel) [inline ms-authentication/routes/controller.go:13]:
```suggestion
	service interfaces.ApplicationService
	lc logger.Client
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by brian-intel — CHANGES_REQUESTED** [review]

File: ms-authentication/routes/controller.go (hunk 4 above)
Comment (joshua-silverio) [reply ms-authentication/routes/controller.go:18]: @jim-wang-intel I think has a conflicting opinion any thoughts?

File: ms-authentication/routes/gets.go [hunk 6]
```diff
@@ -99,7 +109,7 @@ func AuthenticationGet(writer http.ResponseWriter, req *http.Request) {
 		// 	utilities.WriteStringHTTPResponse(writer, req, http.StatusInternalServerError, "Failed to return authentication data properly", true)
```
[the remark this answers is not in the document]
Comment (joshua-silverio) [reply ms-authentication/routes/gets.go]: removed

**Review by brian-intel — APPROVED** [review]

File: ms-authentication/routes/controller.go (hunk 4 above)
Comment (jim-wang-intel) [reply ms-authentication/routes/controller.go:18]: i am fine either way, with lenny's suggestion, just another copy of logging client here.

Comment (lenny-intel) [reply ms-authentication/routes/controller.go:18]: This makes the code using the logger cleaner and easier to read. Commonly used approach in the EdgeX code.

File: ms-authentication/routes/controller_test.go [hunk 7]
```diff
@@ -34,10 +35,10 @@ func TestController_AddAllRoutes(t *testing.T) {
 			mockAppService := &mocks.ApplicationService{}
 			if !tt.failAddRoute {
 				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
-					Return(nil)
+					Return(nil).On("LoggingClient").Return(logger.NewMockClient())
 			} else {
 				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).
-					Return(fmt.Errorf("fail"))
+					Return(fmt.Errorf("fail")).On("LoggingClient").Return(logger.NewMockClient())
```
Comment (lenny-intel) [inline ms-authentication/routes/controller_test.go:41]: For readability, please don't chain these. The second mock is not so apparent as when we just have
mockAppService.On("LoggingClient").Return(logger.NewMockClient())
On its own line.

File: ms-authentication/routes/gets_test.go [hunk 8]
```diff
@@ -247,11 +247,9 @@ func TestAuthenticationGet(t *testing.T) {
 		t.Run(currentTest.Name, func(t *testing.T) {
 			mockAppService := &mocks.ApplicationService{}
 			mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).On("LoggingClient").
-				Return(logger.NewMockClient())
+				Return(logger.NewMockClient()).On("LoggingClient").Return(logger.NewMockClient())
```
Comment (lenny-intel) [inline ms-authentication/routes/gets_test.go:250]: Same comment on chaining these.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ms-authentication/routes/controller_test.go (hunk 7 above)
Comment (joshua-silverio) [reply ms-authentication/routes/controller_test.go]: agreed, moved to new line

File: ms-authentication/routes/controller_test.go [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockAppService := &mocks.ApplicationService{}
+			if !tt.failAddRoute {
+				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).
+					On("LoggingClient").Return(logger.NewMockClient())
+			} else {
+				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("fail")).
+					On("LoggingClient").Return(logger.NewMockClient())
+			}
```
Comment (lenny-intel) [inline ms-authentication/routes/controller_test.go:42]:
```suggestion
			mockAppService := &mocks.ApplicationService{}
			mockAppService.On("LoggingClient").Return(logger.NewMockClient()
			
			if !tt.failAddRoute {
				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
			} else {
				mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(fmt.Errorf("fail"))
			}
```

File: ms-authentication/routes/gets_test.go [hunk 10]
```diff
@@ -242,6 +245,12 @@ func TestAuthenticationGet(t *testing.T) {
 	for _, test := range tests {
 		currentTest := test
 		t.Run(currentTest.Name, func(t *testing.T) {
+			mockAppService := &mocks.ApplicationService{}
+			mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).
+				On("LoggingClient").Return(logger.NewMockClient())
```
Comment (lenny-intel) [inline ms-authentication/routes/gets_test.go:250]:
```suggestion
			mockAppService := &mocks.ApplicationService{}
			mockAppService.On("AddRoute", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
			mockAppService.On("LoggingClient").Return(logger.NewMockClient())
				
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ms-authentication/routes/controller_test.go (hunk 7 above)
Comment (lenny-intel) [reply ms-authentication/routes/controller_test.go]: Not what I intended... ;-) See new comments.

File: ms-authentication/routes/gets_test.go (hunk 10 above)
Comment (joshua-silverio) [reply ms-authentication/routes/gets_test.go]: ahh makes sense changed to two lines

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**